client/channel.go

```diff
@@ -85,3 +85,16 @@ func (c *Client) CreateChannel(appID string, appType string, name string, descri
 
 	return nil, errors.New("unknown app type")
 }
+
+func (c *Client) GetChannelByName(appID string, appType string, name string, description string, createIfAbsent bool) (*types.Channel, error) {
+
+	if appType == "platform" {
+		return nil, errors.New("ensure-channel operations are not supported for platform applications")
+	} else if appType == "ship" {
+		return nil, errors.New("ensure-channel operations are not supported for ship applications")
```

Comment:
do we not have GetChannelByName without the ensure-channel bits for these schedulers?

Reply:
this isn't a regression though, so not a blocker

Reply:
Just, didn't feel like implementing it. This is partially because the current code structure requires a lot of code dupe to do so